packages/@aws-cdk/aws-iam/lib/policy-document.ts

```diff
@@ -24,6 +24,20 @@ export interface PolicyDocumentProps {
  * A PolicyDocument is a collection of statements
  */
 export class PolicyDocument implements cdk.IResolvable {
+
+  /**
+   * Creates a new PolicyDocument based on the object provided.
+   * This will accept an object created from the `.toJSON()` call
+   * @param obj the PolicyDocument in object form.
+   */
+  public static fromJson(obj: any): PolicyDocument {
+    const newPolicyDocument = new PolicyDocument();
+    if (obj.Statement && Array.isArray(obj.Statement)) {
```

Comment:
Should we fail if `obj.Statement` is not an array?

Reply:
Perhaps... some explicit checking of Statement and other fields could be of value.

Reply:
added an explicit test

Reply:
Coding style: we prefer "early bailout" so the "happy flow" of the function is kept at the top-level indentation. Additionally, try to avoid "if" statements by using default values. 

Here is a sketch:

```ts
const statement = obj.Statement ?? [ ];
if (statement && !Array.isArray(statement) {
  throw new Error(`...`);
}

// here I would just call the variable `s`. It's pretty clear what it is.
newPolicyDocument.addStatements(...statements.map(s => PolicyDocument.fromJson(s));
```

Reply:
updated...